packages/py-moose-lib/moose_lib/streaming/streaming_function_runner.py

```diff
@@ -0,0 +1,512 @@
+    """
+    Load a DMV1 (legacy) streaming function from a Python module.
+    
+    Args:
+        function_file_dir: Directory containing the streaming function module
+        function_file_name: Name of the module file without .py extension
+        
+    Returns:
+        Tuple of (input_type, run_function) where:
+            - input_type is the type annotation of the run function's input parameter
+            - run_function is the actual transformation function
+            
+    Raises:
+        SystemExit: If module import fails or if multiple/no streaming functions found
+    """
+    sys.path.append(function_file_dir)
+
+    try:
+        # todo: check the flat naming
+        module = import_module(function_file_name)
+        streaming_function_def = module.StreamingFunction
+    except Exception as e:
+        cli_log(CliLogData(action="Function", message=str(e), message_type="Error"))
+        sys.exit(1)
+
+    # Get all the named flows in the flow file and make sure the flow is of type StreamingFunction
+    streaming_functions = [f for f in dir(module) if isinstance(getattr(module, f), streaming_function_def)]
+
+    # Make sure that there is only one flow in the file
+    if len(streaming_functions) != 1:
+        cli_log(CliLogData(action="Function", message=f"Expected one streaming function in the file, but got {len(streaming_functions)}", message_type="Error"))
+        sys.exit(1)
+
+    # get the flow definition
+    streaming_function_def = getattr(module, streaming_functions[0])
+
+    # get the run function
+    streaming_function_run = streaming_function_def.run
+
+    # get run input type that doesn't rely on the name of the input parameter
+    run_input_type = streaming_function_run.__annotations__[list(streaming_function_run.__annotations__.keys())[0]]
+
+    return run_input_type, streaming_function_run
+
+def load_streaming_function_dmv2(function_file_dir: str, function_file_name: str) -> tuple[type, Callable]:
+    """
+    Load a DMV2 streaming function by finding the stream transformation that matches
+    the source and target topics.
+    
+    Args:
+        function_file_dir: Directory containing the main.py file
+        function_file_name: Name of the main.py file (without extension)
+        
+    Returns:
+        Tuple of (input_type, transformation_function) where:
+            - input_type is the Pydantic model type of the source stream
+            - transformation_function is the function that transforms source to target data
+            
+    Raises:
+        SystemExit: If module import fails or if no matching transformation is found
+    """
+    # Add the function directory to Python path so we can import the main module
+    sys.path.append(function_file_dir)
+    
+    try:
+        # Import the main module which should define the streams
+        import_module(function_file_name)
+        module = import_module("moose_lib.dmv2")
```

Comment:
This can be on the top, now that it's in moose_lib?
`from moose_lib.dmv2 import _streams`